sound/soc/sof/intel/hda.c

```diff
@@ -1065,7 +1065,10 @@ int hda_dsp_remove(struct snd_sof_dev *sdev)
 				SOF_HDA_INT_CTRL_EN | SOF_HDA_INT_GLOBAL_EN, 0);
 
 	/* disable cores */
-	if (chip)
+	if (chip && chip->hw_ip_version == SOF_INTEL_ACE_1_0) {
+		if (sof_ops(sdev)->core_put)
+			sof_ops(sdev)->core_put(sdev, chip->host_managed_cores_mask);
+	} else if (chip)
```

Comment:
I think the simplest fix is to create a new mtl_dsp_remove() function that calls mlt_core_power_down instead of hda_dsp_core_reset_power_down.

Reply:
I have another branch has mtl_dsp_remove(), a few functions need to to be modified also. Let me upload that change here.